netman/adapters/switches/juniper/base.py

```diff
@@ -324,6 +324,23 @@ def set_interface_native_vlan(self, interface_id, vlan):
                     raise UnknownVlan(vlan)
                 raise
 
+    def set_interface_auto_negotiation_state(self, interface_id, negotiation_state):
+        content = to_ele("""
+            <interface>
+                <name>{0}</name>
+            </interface>
+            """.format(interface_id))
+        if negotiation_state == ON:
+            content.append(to_ele("<ether-options><auto-negotiation></ether-options>"))
+        elif negotiation_state == OFF:
+            content.append(to_ele("<ether-options><no-auto-negotiation></ether-options>"))
+        else:
+            raise ValueError("Unknown negotiation_state {}, expected [ON,OFF]", negotiation_state)
```

Comment:
This parameter is "an enum" and is part of the contract to be ON or OFF and is enforced by the API, you don't need to handle this case here